keepsorted/cmp.go

```diff
@@ -0,0 +1,33 @@
+package keepsorted
+
+import (
+	"cmp"
+)
+
+type cmpFunc[T any] func(T, T) int
+
+func (f cmpFunc[T]) andThen(next cmpFunc[T]) cmpFunc[T] {
```

Comment:
Can you add comments to the functions of this file? Especially this method. I had to read it carefully to think about what it meant. But also I suspect the average Go programmer is not gonna be well acquainted with generics.